android-file-chooser/src/main/java/ir/sohreco/androidfilechooser/FileChooserDialog.java

```diff
@@ -309,7 +308,7 @@ public void onClick(View view) {
         int id = view.getId();
         if (id == R.id.previous_dir_imagebutton) {
             File parent = new File(currentDirectoryPath).getParentFile();
-            if (parent != null && parent.canRead()) {
+            if (parent != null) {
```

Comment:
I think there is no point in doing this. Doing a null check and checking if the app can access a directory are the same in this situation. In your code, if the app is not allowed to access a directory then listFiles method returns null and nothing will be shown to the user. My code achieves the same result. If a directory is readable then its files will be listed and if it's not then the code will not try to list it's files.